src/passes/OptimizeInstructions.cpp

```diff
@@ -1773,6 +1785,67 @@ struct OptimizeInstructions
     }
   }
 
+  BinaryOp makeUnsignedBinaryOp(BinaryOp op) {
+    switch (op) {
+      case DivSInt32:
+        return DivUInt32;
+      case RemSInt32:
+        return RemUInt32;
+      case ShrSInt32:
+        return ShrUInt32;
+      case LtSInt32:
+        return LtUInt32;
+      case LeSInt32:
+        return LeUInt32;
+      case GtSInt32:
+        return GtUInt32;
+      case GeSInt32:
+        return GeUInt32;
+
+      case DivSInt64:
+        return DivUInt64;
+      case RemSInt64:
+        return RemUInt64;
+      case ShrSInt64:
+        return ShrUInt64;
+      case LtSInt64:
+        return LtUInt64;
+      case LeSInt64:
+        return LeUInt64;
+      case GtSInt64:
+        return GtUInt64;
+      case GeSInt64:
+        return GeUInt64;
+
+      default:
+        return InvalidBinary;
+    }
+  }
+
+  bool isSignedBinaryOp(BinaryOp op) {
+    switch (op) {
+      case DivSInt32:
+      case RemSInt32:
+      case ShrSInt32:
+      case LtSInt32:
+      case LeSInt32:
+      case GtSInt32:
+      case GeSInt32:
+
+      case DivSInt64:
+      case RemSInt64:
+      case ShrSInt64:
+      case LtSInt64:
+      case LeSInt64:
+      case GtSInt64:
+      case GeSInt64:
+        return true;
+
+      default:
+        return false;
+    }
+  }
```

Comment:
I don't think this function is needed anymore, right?

Reply:
right!